packages/components/src/snackbar/index.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import { flow, noop } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '../';
+
+const stopPropagation = ( event ) => event.stopPropagation();
+const NOTICE_TIMEOUT = 10000;
+
+function Snackbar( {
+	className,
+	children,
+	actions = [],
+	onRemove = noop,
+} ) {
+	useEffect( () => {
+		// This rule doesn't account yet for React Hooks
+		// eslint-disable-next-line @wordpress/react-no-unsafe-timeout
+		const timeoutHandle = setTimeout( () => {
+			onRemove();
+		}, NOTICE_TIMEOUT );
+
+		return () => clearTimeout( timeoutHandle );
+	}, [] );
```

Comment:
Doesn't have to be done in this pull, but one thing that could get around having to use the effect rule exclusion here is to create a `useTimeout` hook that components like this can use (and the only place the es-lint rule would need  skipped is in with that custom hook definition).

Reply:
Interesting, I can see how this can be useful. Maybe we could try it once we find a second use-case?